sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1220,19 +1213,11 @@ class DataFrame private[sql](
    * @since 1.3.0
    */
   def withColumnRenamed(existingName: String, newName: String): DataFrame = {
-    val resolver = sqlContext.analyzer.resolver
     val output = queryExecution.analyzed.output
-    val shouldRename = output.exists(f => resolver(f.name, existingName))
-    if (shouldRename) {
-      val columns = output.map { col =>
-        if (resolver(col.name, existingName)) {
-          Column(col).as(newName)
-        } else {
-          Column(col)
-        }
-      }
-      select(columns : _*)
-    } else {
+    resolveToIndex(existingName).map {index =>
+      select(output.map(attr =>
+        Column(attr)).updated(index, Column(output(index)).as(newName)) : _*)
```

Comment:
we can define a `val renamed = Column(output(index)).as(newName)` first and make this line short enough to fit in one line